static/app/views/projectInstall/issueAlertNotificationOptions.tsx

```diff
@@ -0,0 +1,150 @@
+import {useEffect, useMemo, useState} from 'react';
+import styled from '@emotion/styled';
+
+import MultipleCheckbox from 'sentry/components/forms/controls/multipleCheckbox';
+import {space} from 'sentry/styles/space';
+import {IssueAlertActionType} from 'sentry/types/alerts';
+import type {OrganizationIntegration} from 'sentry/types/integrations';
+import {useApiQuery} from 'sentry/utils/queryClient';
+import useOrganization from 'sentry/utils/useOrganization';
+import SetupMessagingIntegrationButton, {
+  MessagingIntegrationAnalyticsView,
+} from 'sentry/views/alerts/rules/issue/setupMessagingIntegrationButton';
+import {useIssueAlertNotificationContext} from 'sentry/views/projectInstall/issueAlertNotificationContext';
+import MessagingIntegrationAlertRule from 'sentry/views/projectInstall/messagingIntegrationAlertRule';
+
+export const providerDetails = {
+  slack: {
+    name: 'Slack',
+    action: IssueAlertActionType.SLACK,
+    label: 'workspace to',
+    placeholder: 'channel, e.g. #critical',
+  },
+  discord: {
+    name: 'Discord',
+    action: IssueAlertActionType.DISCORD,
+    label: 'server in the channel',
+    placeholder: 'channel ID or URL',
+  },
+  msteams: {
+    name: 'MSTeams',
+    action: IssueAlertActionType.MS_TEAMS,
+    label: 'team to',
+    placeholder: 'channel ID',
+  },
+};
+
+enum MultipleCheckboxOptions {
+  EMAIL = 'email',
+  INTEGRATION = 'integration',
+}
+
+export default function IssueAlertNotificationOptions() {
+  const organization = useOrganization();
+  const {
+    alertNotificationAction: action,
+    alertNotificationProvider: provider,
+    setAlertNotificationAction: setAction,
+    setAlertNotificationIntegration: setIntegration,
+    setAlertNotificationProvider: setProvider,
+  } = useIssueAlertNotificationContext();
```

Comment:
Looks like this context is __only__ used once here. Do we really need a context?

Typically you want a context when you need state shared a cross indeterminately nested components where you wouldn't want to have to prop drill constantly

Reply:
I wanted to avoid prop drilling from `createProject`,  the context is being used in both `issueAlertNotificationOptions` and the nested `messagingIntegrationAlertRule` component